app/controllers/surveys_controller.rb

```diff
@@ -80,4 +81,9 @@ def require_ownership
         redirect_to root_path, notice: "You are not the owner of that survey or an admin"
       end
     end
+
+    def require_published
+      redirect_to root_path, notice: 'Survey not published.' if !@survey.published?
```

Comment:
will this work? on the right track though...

Reply:
Ah, oops! I just spotted what is wrong. I think brackets might help here?